### PR TITLE
Improved module search

### PR DESCRIPTION
**Pull request by aafemt**
1) It can find not only "module" and "libmodule.so", but also "module.so".
2) No second attempt if doctor made no changes in name.

### CODE REVIEW

File: src/common/os/posix/mod_loader.cpp [hunk 1]
```diff
@@ -69,25 +69,39 @@ bool ModuleLoader::isLoadableModule(const Firebird::PathName& module)
 	return true;
 }
 
-void ModuleLoader::doctorModuleExtension(Firebird::PathName& name)
+bool ModuleLoader::doctorModuleExtension(Firebird::PathName& name, int& step)
```
Comment (AlexPeshkoff) [inline src/common/os/posix/mod_loader.cpp]: I like an idea of patch but would not it be better to use
class ModuleName : public PathName {.......
containing step number instead adding artificial parameter to doctorModuleExtension function which must be in sync with module's PathName?

**Comment by aafemt** [comment]
> I like an idea of patch but would not it be better to use
> class ModuleName : public PathName {.......
> containing step number instead adding artificial parameter to doctorModuleExtension 
> function which must be in sync with module's PathName?

No. It would make only troubles and no advatages. I won't jump into this trap again.

**Comment by AlexPeshkoff** [comment]
One advantage is obvious - related data is stored in single data structure and there is no chance to mix unrelated module name and doctor's step.
What troubles do you see?

**Comment by aafemt** [comment]
fixAndLoad() should return fixed name. Here are two possibilities:
1) A tiny class with single member "step" and reference/pointer to PathName which is used in exactly two places. Too little result for such effort: this change will increase amount of code without increasing of functionality.
2) Snowball of type changes starting from fixAndLoad's parameter rolling over whole call stack. This gocha I already had with Unicode branch, so - no, thanks.

**Comment by AlexPeshkoff** [comment]
OK, let's try to choose minimum evil

**Pull request merged**